### PR TITLE
Silence deprecation warnings inside the library itself

### PR DESCRIPTION
**Pull request by randombit**
GH #1243

### CODE REVIEW

**Comment by webmaster128** [comment]
Cool, thanks. Can we use this in the other internal uses of BOTAN_NO_DEPRECATED_WARNINGS as well? This would prevent suppressing other deprecations in those source files.

```
src/lib/tls/tls_blocking.cpp-This API is itself deprecated, so we don't care that it relies on
src/lib/tls/tls_blocking.cpp-other deprecated things
src/lib/tls/tls_blocking.cpp-*/
src/lib/tls/tls_blocking.cpp:#define BOTAN_NO_DEPRECATED_WARNINGS
src/lib/tls/tls_blocking.cpp-
src/lib/tls/tls_blocking.cpp-#include <botan/tls_blocking.h>
src/lib/tls/tls_blocking.cpp-
--
src/tests/test_filters.cpp-* Botan is released under the Simplified BSD License (see license.txt)
src/tests/test_filters.cpp-*/
src/tests/test_filters.cpp-
src/tests/test_filters.cpp:#define BOTAN_NO_DEPRECATED_WARNINGS
src/tests/test_filters.cpp-
src/tests/test_filters.cpp-#include "tests.h"
src/tests/test_filters.cpp-
--
src/tests/test_package_transform.cpp-* Botan is released under the Simplified BSD License (see license.txt)
src/tests/test_package_transform.cpp-*/
src/tests/test_package_transform.cpp-
src/tests/test_package_transform.cpp:#define BOTAN_NO_DEPRECATED_WARNINGS
src/tests/test_package_transform.cpp-
src/tests/test_package_transform.cpp-#include "tests.h"
src/tests/test_package_transform.cpp-
--
src/tests/unit_tls.cpp-*/
src/tests/unit_tls.cpp-
src/tests/unit_tls.cpp-// Deprecated TLS APIs are tested below
src/tests/unit_tls.cpp:#define BOTAN_NO_DEPRECATED_WARNINGS
src/tests/unit_tls.cpp-
src/tests/unit_tls.cpp-#include "tests.h"
src/tests/unit_tls.cpp-#include <vector>
```

File: src/lib/tls/tls_blocking.cpp [hunk 1]
```diff
@@ -6,12 +6,6 @@
 * Botan is released under the Simplified BSD License (see license.txt)
 */
 
-/*
-This API is itself deprecated, so we don't care that it relies on
-other deprecated things
```
Comment (webmaster128) [inline src/lib/tls/tls_blocking.cpp]: I found this comment useful. Can we move it down to the use of `TLS::Compat_Callbacks(TLS::Compat_Callbacks::SILENCE_DEPRECATION_WARNING::PLEASE, …`?

**Comment by randombit** [comment]
For the tests it didn't seem to matter much because the amalgamation doesn't affect those.

**Comment by randombit** [comment]
I notice the TLS command line utils are still using the old style constructors, these should actually get fixed to use the Callback based interface.

**Pull request merged**